### DISCUSSION TITLE
Nanostring CosMx and AtoMx exported data

### DISCUSSION THREAD

**Discussion by jmodlis** [giotto-suite/Giotto#783] [Ideas]
Hello, 

Thank you for all your great work on this package! We have been using it extensively on some older Nanostring CosMx datasets. We are now working with Nanostring CosMx data that has been minimally processed and exported from the Nanostring AtoMx analysis suite. Unfortunately, the exported data structure from AtoMx is quite different from what the Giotto function `createGiottoCosMxObject` expects.  Are there any plans to accommodate this new structure into Giotto functions? I'm working on work-arounds myself but thought it would be good to see if others were facing this problem as well.

**Comment by RubD** [comment]
hi @jmodlis Can you share how the structure looks?

Although `createGiottoCosMxObject` might not work, it's really just a wrapper around some lower level functions that should work on virtually any type of input data. You might need to figure out how to set some of the params. We're currently pretty busy with implementing some new functionality, but could take a look to see how easy it would be to provide another wrapper.

**Comment by simonmelov** [comment]
That would be great, we are faced with a similar issue

**Reply by RubD** [reply]
Can you share a small example, or if you're working with someone from Nanostring, let them reach out to us? Thanks

**Comment by simonmelov** [comment]
Hi,
Where do you want me to send?
Best
S

**Reply by RubD** [reply]
Can you put the data in a dropbox/gdrive/onedrive/... folder and share a link with me (rdries @ bu.edu)?

**Comment by jmodlis** [comment]
Hello,

I can also share my code for what progress I have made thus far.  At this point I have created a hacked solution that involves:
- using bash to create symlinks to the images so that the Giotto expectations for the image structure are met. The images from AtoMx are in folders, but some are separated by FOV rather than type.
- importing the AtoMx exported Seurat object and exporting csv files structured as Giotto expects. I wasn't able to find the data needed by Giotto in text or csv files, but almost all of it is found in the Seurat object. 
- the last issue (that I am aware of) is that the FOV positions are supplied locally as pixels and globally as millimeters. So I am solving this by fitting a linear model for each FOV-x and FOX-y to find the intercept (aka shift) and slope (multiplication factor, identical across all FOVs in the one sample I am working with)

Thanks,
Jen

**Reply by RubD** [reply]
Thanks for sharing Jen. Hopefully AtoMx also has a more generic way of exporting files. We're working on improving a seurat to giotto function, but I don't think that will be a very good or stable option long term.

**Comment by RubD** [comment]
@simonmelov and @jmodlis have your issues been resolved? We've worked a lot on improving the code and documentation (www.giottosuite.com) and are now taking a more systematic look at some of the key issues users have faced. 

I'm also looping in @jiajic who has been removing bugs and streamlining code at record pace

**Comment by simonmelov** [comment]
Will re-examine over the weekend! Thanks!

<https://github.com/simonmelov>
This Message originated outside your organization.
________________________________
@simonmelov and @jmodlis<https://github.com/jmodlis> have your issues been resolved? We've worked a lot on improving the code and documentation (www.giottosuite.com<http://www.giottosuite.com>) and are now taking a more systematic look at some of the key issues users have faced.

I'm also looping in @jiajic<https://github.com/jiajic> who has been removing bugs and streamlining code at record pace

**Comment by jmodlis** [comment]
Hello! Thank you for checking in!  Nanostring has now developed a new AtoMx export module that will allow us to export "flat" files, rather than Seurat objects. This is a recent development and we are working on getting the module running on our system. I don't yet have access to the new "flat" files myself.  I do know that they are supposed to be loadable via Seurat's `LoadNanostring` function.  I think I should have access to the files in the next week or so.  I was eventually able to get the AtoMx exported data loaded into a fully analyze-able Giotto object.  I will do my best to keep you posted. Thanks! Jen

**Comment by simonmelov** [comment]
Here ya go!
Best
S

This Message originated outside your organization.
________________________________
Hello! Thank you for checking in! Nanostring has now developed a new AtoMx export module that will allow us to export "flat" files, rather than Seurat objects. This is a recent development and we are working on getting the module running on our system. I don't yet have access to the new "flat" files myself. I do know that they are supposed to be loadable via Seurat's LoadNanostring function. I think I should have access to the files in the next week or so. I was eventually able to get the AtoMx exported data loaded into a fully analyze-able Giotto object. Thanks! Jen

**Comment by simonmelov** [comment]
As of ver 1.3 in AtomX, the export module no longer outputs the CellComposite directory. Those images are not available anywhere, and Giotto requires this directory to be able to import CosMx data. Is there a workaround for this? Looks like Giotto will not work with CosMx data going forward. With the huge increase in transcripts being measured by CosMx (6000, and full genome coming), this is a shame.

**Reply by jmodlis** [reply]
This is bad... I do think that this particular problem extends beyond Giotto. Other packages (e.g. squidpy) also make use of the composite images. I think the most effective solution may be to bring this up with Nanostring.  I will see what I can do, but it may take feedback from multiple customers.

**Reply by RubD** [reply]
@jmodlis or @simonmelov do you have a link to a page where we can find information about the new output files/format? If we can also download the new data, then that would also be very helpful. The convenience function might be broken but you should still be able to create a giotto object by reading in the necessary data one by one (e.g. transcript and polygon coordinates).

**Reply by rbutleriii** [reply]
Check the `4 High Res FOV FL Image` folder. I wasn't using the AtomMx portal, but my latest slides they were in that folder, not the `5 Raw data` folder. Basically you need to move all the stuff in 4 -> 5 to recreate the full set as before.

**Reply by rbutleriii** [reply]
Also, heads up. #946 may also be in your future...

**Reply by simonmelov** [reply]
They are putting it back soon!

**Comment by andrewrech** [comment]
> I will see what I can do, but it may take feedback from multiple customers.

I am following this and will contribute. Thanks

**Comment by jiajic** [comment]
Hi all, we have been working on an update for the CosMx convenience function on the `@modular_readers` branch of Giotto and put together an import utility that will probably be thinly wrapped in the final convenience function.

It should be a more flexible and modular approach to loading where, if necessary, you can load in more or less of a particular data type by just pointing it towards the requisite filepaths or subdirectories.

It also addresses some issues, such as different naming between the Nanostring outputs vs Giotto's generated IDs and avoiding the usage of `joinGiottoObjects()` during object creation.

Please take a look at this [gist](https://gist.github.com/jiajic/bc8abf18fd362ca36790b8c074847bb9) for further introduction. Any feedback would be welcome!

Best,
George

**Reply by rbutleriii** [reply]
Will try to take a look next week with some test data, thanks!
